src/components/Task/Modal/TaskModal.tsx

```diff
@@ -45,69 +52,63 @@ const TaskModal = ({
 
   const task = data?.task;
 
-  const renderTitle = (): string => {
+  const renderTitle = useMemo((): string => {
     switch (view) {
-      case 'complete':
+      case TaskModalEnum.Complete:
         return t('Complete Task');
-      case 'comments':
+      case TaskModalEnum.Comments:
         return t('Task Comments');
-      case 'log':
+      case TaskModalEnum.Log:
         return t('Log Task');
-      case 'edit':
+      case TaskModalEnum.Edit:
         return t('Edit Task');
       default:
         return t('Add Task');
     }
-  };
+  }, [view]);
 
-  const renderView = (): ReactElement => {
-    switch (view) {
-      case 'complete':
-        if (task) {
-          return (
+  return loading ? (
+    <Loading loading />
+  ) : (
+    <Modal isOpen={open} title={renderTitle} handleClose={onModalClose}>
+      {accountListId ? (
+        <>
+          {view === TaskModalEnum.Complete && task && (
             <DynamicTaskModalCompleteForm
               accountListId={accountListId || ''}
```

Comment:
Now that you're checking `accountListId` on line 74, we can simplify this and the three other occurrences.

```suggestion
              accountListId={accountListId}
```